openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,191 @@
+
+    def _execute_post(self, url, data=None, headers=None, json_data=None):
+        """
+        Execute post request to achieve the ICC external enrollment.
+        """
+        return requests.post(
+            url=url,
+            data=json_data,
+        )
+
+    def _get_enrollment_headers(self):
+        """
+        Method that returns None by default, ICC integration does not require headers.
+        """
+        return None
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        Method that provide the relevant data for the enrollment.
+        """
+        enrollment_data = {}
+
+        try:
+            enrollment_data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_ENROLLMENT_API_FUNCTION,
+                "enrolments[0][roleid]": settings.ICC_LEARNER_ROLE_ID,
+                "enrolments[0][userid]": self._get_icc_user(data).get("id"),
+                "enrolments[0][courseid]": course_settings.get("external_course_run_id"),
+            }
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+        return enrollment_data
+
+    def _get_enrollment_url(self, course_settings):
+        """
+        Method that returns base url for the ICC integration API.
+        """
+        return settings.ICC_BASE_URL
+
+    def _get_icc_user(self, data):
+        """
+        Method that look for the user in ICC database through the API, if the user is not
+        found then it calls create method.
+        """
+        log_details = {
+            "url": settings.ICC_API_TOKEN,
+        }
+        icc_user = {}
+
+        try:
+            request_data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_GET_USER_API_FUNCTION,
+                "criteria[0][key]": "email",
+                "criteria[0][value]": data.get("user_email"),
+            }
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=request_data,
+            )
+            log_details["request_payload"] = request_data
+        except Exception as error:  # pylint: disable=broad-except
+            LOG.error("Failed to retrieve ICC user. Reason: %s", str(error))
+            log_details["request_payload"]["wstoken"] = "icc-api-token"
+            log_details["response"] = {"error": "Failed to retrieve ICC user. Reason: " + str(error)}
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, "get_user")
+            icc_user = _validate_icc_user(icc_user)
+
+        return icc_user
+
+    def _create_icc_user(self, data, duplicated_username):
+        """
+        Method that creates a user in the ICC database based in the  current user logged in data.
```

Comment:
One last thing, please remove the double space in "... in the  current user...". @ivanvgh

Reply:
Done.